drivers/python/rethinkdb/net_twisted.py

```diff
@@ -92,42 +80,39 @@ def _handleHandshake(self, data):
                     self.wait_for_handshake.callback(None)
                     # We're now ready to work with real data.
                     self.state = DatabaseProtocol.READY
-                    self.resetBuffer()
         except Exception as e:
             self.wait_for_handshake.errback(e)
 
     def _handleResponse(self, data):
-        # 1. Read the header, until we read the length of the awaited payload.
-        if self.buf_expected_length == 0:
-            if data >= 12:
-                token, length = struct.unpack('<qL', data[:12])
-                self.buf_token = token
-                self.buf_expected_length = length
-                data = data[12:]
-            else:
-                self.buf += data
-                # We quit the function, it is impossible to have read the
-                # entire payload at this point.
```

Comment:
Rather than recursively handle multiple requests, do it in a loop - no reason to blow the stack in some exceptional cases.